spec/src/provider.spec.js

```diff
@@ -101,6 +101,93 @@ describe("Provider", function() {
     });
   });
 
+  it("rejects method calls when the provider errors", function (done) {
+    var iface = port.getInterface();
+    var o = function() {};
+    o.prototype.m1 = function(str) {
+      throw new Error('hi.myerr');
+    };
+
+    iface.providePromises(o);
+    port.onMessage('default', {
+      channel: 'message'
+    });
+
+    port.onMessage('default', {to: 'testInst', type:'message', message:{
+      'type': 'construct',
+    }});
+
+    port.on('message', function(n) {
+      expect(n.message.error).toContain('hi.myerr');
```

Comment:
Is there more of an end-to-end way to test this? Does this expect guarantee that the Promise will reject?